### PR TITLE
Fix query unit test, wait until write journal is ready

### PR DESCRIPTION
**Pull request by Arkatufus**
## Changes
Wait for the write journal to be ready before starting unit test

### CODE REVIEW

File: src/Akka.Persistence.Sql.Tests.Common/Query/BaseAllEventsSpec.cs [hunk 1]
```diff
@@ -5,25 +5,42 @@
...
             ReadJournal = Sys.ReadJournalFor<SqlReadJournal>(SqlReadJournal.Identifier);
         }
 
+        public async Task InitializeAsync()
+        {
+            // Force start read journal
+            var journal = Persistence.Instance.Apply(Sys).JournalFor(null);
+            
+            // Wait until journal is ready
+            var _ = await journal.Ask<Initialized>(IsInitialized.Instance).ShouldCompleteWithin(3.Seconds());
```
Comment (Aaronontheweb) [inline src/Akka.Persistence.Sql.Tests.Common/Query/BaseAllEventsSpec.cs:38]: LGTM

File: src/Akka.Persistence.Sql/Journal/SqlWriteJournal.cs [hunk 2]
```diff
@@ -137,13 +137,18 @@ private bool Initializing(object message)
...
-                _writeInProgress.Remove(wf.PersistenceId);
-
-            return true;
+            switch (message)
+            {
+                case WriteFinished wf:
+                    if (_writeInProgress.TryGetValue(wf.PersistenceId, out var latestPending) & (latestPending == wf.Future))
+                        _writeInProgress.Remove(wf.PersistenceId);
+                    return true;
+                case IsInitialized:
```
Comment (Aaronontheweb) [inline src/Akka.Persistence.Sql/Journal/SqlWriteJournal.cs:150]: Should add a comment here explaining why this message exists

**Review by Aaronontheweb — CHANGES_REQUESTED** [review]
One small change

File: src/Akka.Persistence.Sql/Journal/SqlWriteJournal.cs (hunk 2 above)
Comment (Arkatufus) [reply src/Akka.Persistence.Sql/Journal/SqlWriteJournal.cs:150]: Good idea

Comment (Arkatufus) [reply src/Akka.Persistence.Sql/Journal/SqlWriteJournal.cs:150]: Done

**Review by Aaronontheweb — APPROVED** [review]
LGTM

**Pull request merged**